install/ui/src/freeipa/widgets/LoginScreenBase.js

```diff
@@ -285,13 +285,17 @@ define(['dojo/_base/declare',
          */
         display_caps_warning: function(display) {
 
-            this.caps_warning = display;
             var val_summary = this.get_widget('validation');
             if (display) {
-                val_summary.add_warning('caps', this.caps_warning_msg);
+                if (!this.caps_warning) {
+                    val_summary.add_warning('caps', this.caps_warning_msg);
+                }
             } else {
```

Comment:
It is enough to add just one condition in the first line of the function:
`if (this.caps_warning == display) return;`
So function will do nothing if the state isn't changed.

Reply:
It is not exactly same.
about 'caps_warning': 
```js
        /**                                                                     
         * Indicates that CAPS LOCK warning is on. Null indicates that we don't 
         * know the state.                                                      
         * @property {boolean|null}                                             
         */                                                                     
        caps_warning: null, 
```
But 'display' is boolean.
If we just use 'if (this.caps_warning == display) return;' under 'display=false' and 'caps_warning=null' then removing of 'caps' validation will be triggered. This is not i want.

Reply:
So
`if (!!this.caps_warning === display) return;`

Reply:
Nice trick :-)
But in this case CapsStatus ('caps_warning') will be catched on 'caps on -> keypress' only.
An expected (current) behavior - on any keypress regardless whether caps is on or off.
Reason of error:
'caps_warning' should become false/true on very first calling of function 'display_caps_warning'.

Reply:
Ok, I agree.